.github/workflows/release-candidate.yml

```diff
@@ -0,0 +1,107 @@
+name: Release Candidate
+
+on:
+  # Manual Trigger GH CLI -> gh workflow run release-candidate.yml --ref <branch-name>
+  workflow_dispatch:
+
+jobs:
+  run-python-tests:
+    uses: ./.github/workflows/python-versions.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-javascript-tests:
+    uses: ./.github/workflows/js-tests.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-py-prod-deps-smoke-test:
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  run-cypress-tests:
+    uses: ./.github/workflows/cypress.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  build-release-candidate:
+    runs-on: ubuntu-latest
+
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
```

Comment:
Hm, while we might want to still upload the nightly package on a test failure (it's likely to be useful anyway), I think we almost definitely want to gate this on everything passing.

Reply:
Currently gated as all `needs:` jobs must complete successfully before this runs ([link confirming this behavior](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)) 👍🏼